test/workflow-orchestration.ts

```diff
@@ -138,6 +138,70 @@ describe('Workflow chaining for a collection configured for swot reprojection an
+    };
+
+    hookRangesetRequest('1.0.0', collection, 'all', { query: reprojectAndZarrQuery });
+    hookRedirect('joe');
+
+    describe('when checking for a query-cmr work item', function () {
+      it('finds and completes it, queueing the next work item for each result', async function () {
+        const res = await getWorkForService(this.backend, 'harmonyservices/query-cmr:latest');
+        expect(res.status).to.equal(200);
+        const workItem = JSON.parse(res.text);
+        expect(workItem.serviceID).to.equal('harmonyservices/query-cmr:latest');
+        workItem.status = WorkItemStatus.SUCCESSFUL;
+        workItem.results = [
+          'test/resources/worker-response-sample/catalog0.json',
+          'test/resources/worker-response-sample/catalog1.json',
+          'test/resources/worker-response-sample/catalog2.json',
+        ];
+        await updateWorkItem(this.backend, workItem);
+        // since there were multiple query cmr results,
+        // multiple work items should be generated for the next step
+        const currentWorkItems = (await getWorkItemsByJobId(db, workItem.jobID)).workItems;
+        expect(currentWorkItems.length).to.equal(4);
+        expect(currentWorkItems.filter((i) => i.status === WorkItemStatus.READY && i.serviceID === 'sds/swot-reproject:latest').length).to.equal(3);
+      });
+
+      describe('when the first swot-reprojection service work item fails', function () {
+        it('the job fails, and all further work items are canceled', async function () {
+          const res = await getWorkForService(this.backend, 'sds/swot-reproject:latest');
+          expect(res.status).to.equal(200);
+          const workItem = JSON.parse(res.text);
+          workItem.status = WorkItemStatus.FAILED;
+          workItem.results = [];
+          await updateWorkItem(this.backend, workItem);
+          expect(workItem.serviceID).to.equal('sds/swot-reproject:latest');
+          // work item failure should trigger job failure
+          const job = await Job.byJobID(db, workItem.jobID);
+          expect(job.status === JobStatus.FAILED);
+          // job failure should trigger cancellation of any pending work items
+          const currentWorkItems = (await getWorkItemsByJobId(db, workItem.jobID)).workItems;
+          expect(currentWorkItems.length).to.equal(4);
+          expect(currentWorkItems.filter((i) => i.status === WorkItemStatus.SUCCESSFUL && i.serviceID === 'harmonyservices/query-cmr:latest').length).to.equal(1);
+          expect(currentWorkItems.filter((i) => i.status === WorkItemStatus.CANCELED && i.serviceID === 'sds/swot-reproject:latest').length).to.equal(2);
+          expect(currentWorkItems.filter((i) => i.status === WorkItemStatus.FAILED && i.serviceID === 'sds/swot-reproject:latest').length).to.equal(1);
+        });
+
+        describe('when checking to see if further reprojection work is queued', function () {
+          hookGetWorkForService('sds/swot-reproject:latest');
```

Comment:
style nit: Feel free to disagree, but I dislike using hooks for a single test (probably have done it myself, though). I think it would be clearer if you just put a call to `getWorkForService` in the `it`. For me hooks make it harder to know what is going on and I only want to use them if they save a lot of redundant code.

Reply:
Actually, looking at the code above (line 177) it would make sense to move this hook up so both of these test cases are using it since they are making the same call.